x-pack/plugins/ml/public/application/jobs/new_job/job_from_pattern_analysis/route_resolver.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import rison from '@kbn/rison';
+import type { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import type { DataPublicPluginStart, TimefilterContract } from '@kbn/data-plugin/public';
+import type { DashboardStart } from '@kbn/dashboard-plugin/public';
+import type { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+import {
+  type CategorizationType,
+  QuickCategorizationJobCreator,
+  CATEGORIZATION_TYPE,
+} from './quick_create_job';
+import type { MlApiServices } from '../../../services/ml_api_service';
+
+import { getDefaultDatafeedQuery } from '../utils/new_job_utils';
+
+interface Dependencies {
+  kibanaConfig: IUiSettingsClient;
+  timeFilter: TimefilterContract;
+  dashboardService: DashboardStart;
+  data: DataPublicPluginStart;
+  mlApiServices: MlApiServices;
+}
+export async function resolver(
+  deps: Dependencies,
+  categorizationTypeRisonString: string,
+  dataViewIdRisonString: string,
+  fieldRisonString: string,
+  partitionFieldRisonString: string | null,
+  stopOnWarnRisonString: string,
+  fromRisonString: string,
+  toRisonString: string,
+  queryRisonString: string
+) {
+  const { mlApiServices, timeFilter, kibanaConfig, dashboardService, data } = deps;
+  let query: QueryDslQueryContainer;
+  try {
+    const queryString = rison.decode(queryRisonString) as string;
+    query = JSON.parse(queryString) as QueryDslQueryContainer;
+  } catch (error) {
+    query = getDefaultDatafeedQuery();
+  }
+
+  let from: string;
+  let to: string;
+  try {
+    from = rison.decode(fromRisonString) as string;
+  } catch (error) {
+    from = '';
+  }
+  try {
+    to = rison.decode(toRisonString) as string;
+  } catch (error) {
+    to = '';
+  }
+
+  let categorizationType;
+  try {
+    categorizationType = rison.decode(categorizationTypeRisonString) as CategorizationType;
+  } catch (error) {
+    categorizationType = CATEGORIZATION_TYPE.COUNT;
```

Comment:
Seems like there's a pattern of catching and setting the value to an empty string - maybe it would be good to have a small util function that abstracts that?

Reply:
Good point, these can be cleaned up quite a lot.
Updated in https://github.com/elastic/kibana/pull/170567/commits/d32017ded1a12c1aa0595d5f5d79dde82080995a
I've also changed the maps and lens versions of this file.